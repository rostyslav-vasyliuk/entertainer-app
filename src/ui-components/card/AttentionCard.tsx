import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { screenWidth } from '../../constants/screen-contants';
import { Icon } from 'native-base';
import { LinearGradient as Gradient } from 'expo-linear-gradient';
const GRADIENT_BLUE = ['#6B84CA', '#8F44CE'];
const GRADIENT_PINK = ['#D442F8', '#B645F5', '#9B40F8'];

const gradientColors = 2 % 2 ? GRADIENT_BLUE : GRADIENT_PINK;

const AttentionCard = () => {
  return (
    <View style={styles.wrapper}>
      <View style={styles.header}>
        <Text style={styles.headerLabel}>
          {'Pay Attention'}
        </Text>
        <Icon name='ios-close' style={styles.closeIcon}/>
      </View>

      <Gradient
          start={[0.45, 0.45]}
          end={[0.90, 0.90]}
          colors={gradientColors}
          style={styles.gradient}
        >
          <Icon name='ios-add-circle' color='white' style={{color: 'white'}}/>
        </Gradient>
    </View>
  );
}

export default AttentionCard;
const BASE_SIZE= 16;
const styles = StyleSheet.create({
  wrapper: {
    height: 100,
    margin: 8,
    width: screenWidth - 16,
    borderRadius: 5,
    borderWidth: 1,
  },
  header: {
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center'
  },
  headerLabel: {
    fontSize: 16,
    fontWeight: '500'
  },
  closeIcon: {
    fontSize: 36,
    padding: 10,
    paddingTop: 0
  },
  gradient: {
    width: BASE_SIZE * 3.25,
    height: BASE_SIZE * 3.25,
    borderRadius: 25,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
