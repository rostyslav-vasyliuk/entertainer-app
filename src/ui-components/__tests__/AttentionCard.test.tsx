import React from 'react';
import AttentionCard from '../card/AttentionCard';
import renderer from 'react-test-renderer';

describe('<AttentionCard />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<AttentionCard />).toJSON();
    expect(tree.children.length).toBe(2);
  });

  it('has 1 child', () => {
    const tree = renderer.create(<AttentionCard />).toJSON();
    expect(tree.children.length).toBeGreaterThanOrEqual(2);
  });

  it('rendered', () => {
    const tree = renderer.create(<AttentionCard />).toJSON();
    expect(tree.children).toBeTruthy();
  });

  it('to match snapshot', () => {
    const tree = renderer.create(<AttentionCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
