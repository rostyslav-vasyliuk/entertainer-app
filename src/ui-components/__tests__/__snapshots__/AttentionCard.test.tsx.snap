// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AttentionCard /> to match snapshot 1`] = `
<View
  style={
    Object {
      "borderRadius": 5,
      "borderWidth": 1,
      "height": 100,
      "margin": 8,
      "width": 734,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      Pay Attention
    </Text>
    <Text />
  </View>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4292100856,
          4290135541,
          4288364792,
        ],
        "endPoint": Array [
          0.9,
          0.9,
        ],
        "locations": undefined,
        "startPoint": Array [
          0.45,
          0.45,
        ],
      }
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 25,
        "height": 52,
        "justifyContent": "center",
        "width": 52,
      }
    }
  >
    <Text />
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
